### ISSUE TITLE
`Swift 6.0` & typed errors - rule `opaqueGenericParameters` provides incorrect results

### ISSUE THREAD

**Issue by denisgaskov** [nicklockwood/SwiftFormat#1845]
## Sample code
This
```swift
func sample<ErrorType>(error: ErrorType) throws(ErrorType) {
  throw error
}
```
is transformed to this (which obviously does not compile):
```swift
func sample(error: some Any) throws(ErrorType) {
  throw error
}
```

## Environment
- SwiftFormat: `0.54.3`
- Xcode: `Version 16.0 beta 6 (16A5230g)`
- Swift: `swiftlang-6.0.0.9.10 clang-1600.0.26.2`

**Comment by nicklockwood** [comment]
Oh interesting - I guess Swift is inferring from the throws() that the type conforms to Error here, even though you don't state so explicitly (cc @calda)

**Comment by nicklockwood** [comment]
@denisgaskov fixed in 0.54.4
